src/Datadog.Trace.DuckTyping/DuckTypeExceptions.cs

```diff
@@ -0,0 +1,172 @@
+    public class DuckTypePropertyOrFieldNotFoundException : DuckTypeException
+    {
+        internal DuckTypePropertyOrFieldNotFoundException(string name)
+            : base($"The property or field  for '{name}' was not found in the instance.")
+        {
+        }
+    }
+
+    /// <summary>
+    /// DuckType type is not an interface exception
+    /// </summary>
+    public class DuckTypeTypeIsNotValidException : DuckTypeException
+    {
+        internal DuckTypeTypeIsNotValidException(Type type, string argumentName)
+            : base($"The type '{type.FullName}' is not a valid type, argument: '{argumentName}'")
+        {
+        }
+    }
+
+    /// <summary>
+    /// DuckType type is not public exception
+    /// </summary>
+    public class DuckTypeTypeIsNotPublicException : DuckTypeException
+    {
+        internal DuckTypeTypeIsNotPublicException(Type type, string argumentName)
+            : base($"The type '{type.FullName}' must be public, argument: '{argumentName}'")
+        {
+        }
+    }
+
+    /// <summary>
+    /// DuckType struct members cannot be changed exception
+    /// </summary>
+    public class DuckTypeStructMembersCannotBeChangedException : DuckTypeException
+    {
+        internal DuckTypeStructMembersCannotBeChangedException(Type type)
+            : base($"Modifying struct members is not supported. [{type.FullName}]")
+        {
+        }
+    }
+
+    /// <summary>
+    /// DuckType target method can not be found exception
+    /// </summary>
+    public class DuckTypeTargetMethodNotFoundException : DuckTypeException
+    {
+        internal DuckTypeTargetMethodNotFoundException(MethodInfo method)
+            : base($"The target method for the proxy method '{method}' was not found.")
+        {
+        }
+    }
+
+    /// <summary>
+    /// DuckType target method parameter is missing exception
+    /// </summary>
+    public class DuckTypeTargetMethodParameterIsMissingException : DuckTypeException
+    {
+        internal DuckTypeTargetMethodParameterIsMissingException(MethodInfo targetMethod, ParameterInfo proxyParameterInfo)
+            : base($"The target method '{targetMethod.Name}' is missing parameter '{proxyParameterInfo.Name}' declared in the proxy method.")
+        {
+        }
+    }
+
+    /// <summary>
+    /// DuckType proxy method parameter is missing exception
+    /// </summary>
+    public class DuckTypeProxyMethodParameterIsMissingException : DuckTypeException
+    {
+        internal DuckTypeProxyMethodParameterIsMissingException(MethodInfo proxyMethod, ParameterInfo targetParameterInfo)
+            : base($"The proxy method '{proxyMethod.Name}' is missing parameter '{targetParameterInfo.Name}' declared in the target method.")
+        {
+        }
+    }
+
+    /// <summary>
+    /// DuckType parameter signature mismatch between proxy and target method
+    /// </summary>
+    public class DuckTypeProxyAndTargetMethodParameterSignatureMismatch : DuckTypeException
```

Comment:
nit: this is missing the `Exception` in the name and should be renamed to `DuckTypeProxyAndTargetMethodParameterSignatureMismatchException`

Reply:
Ups!, good catch... fixed in https://github.com/DataDog/dd-trace-dotnet/pull/946/commits/564fcdf77e48b4645b9881962e5bcd6dd38ee9f5